coremltools/converters/nnssa/coreml/ssa_converter.py

```diff
@@ -806,11 +805,22 @@ def get_transpose_map(prefix, suffix):
                 dim_a = [dim for dim in a if dim not in b][0]
                 dim_b = [dim for dim in b if dim not in a][0]
 
-                # Re-order a and b
-                a_sorted = sorted([dim for dim in a if dim not in [dim_a, dim_reduce]]) + [dim_a, dim_reduce]
-                b_sorted = sorted([dim for dim in b if dim not in [dim_b, dim_reduce]]) + [dim_reduce, dim_b]
-                assert(a_sorted[:-2] == b_sorted[:-2])
-                assert(a_sorted[-1] == b_sorted[-2])
+                a_sorted = []
+                b_sorted = []
+                if len(suffix) == len(a) and len(suffix) == len(b):
+                    #Do the batch mat mul
+                    # Re-order a and b
+                    a_sorted = sorted([dim for dim in a if dim not in [dim_a, dim_reduce]]) + [dim_a, dim_reduce]
+                    b_sorted = sorted([dim for dim in b if dim not in [dim_b, dim_reduce]]) + [dim_reduce, dim_b]
+                    assert(a_sorted[-1] == b_sorted[-2])
+                    assert(a_sorted[:-2] == b_sorted[:-2])
+
+                else:
+                    a_sorted = sorted([dim for dim in a if dim != dim_reduce]) + [dim_reduce]
+                    b_sorted = [dim_reduce] + sorted([dim for dim in b if dim != dim_reduce])
+                    assert(a_sorted[-1] == b_sorted[0])
```

Comment:
If I'm a user, I'd like to understand what's causing the assertion failure, and how could I avoid that. Let's give back some meaningful error messages here.

Reply:
Good!